loglist/logfilter.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loglist
+
+import (
+	"time"
+
+	"github.com/google/certificate-transparency-go/x509"
+
+	"github.com/mohae/deepcopy"
+)
+
+// LogRoots maps Log-URLs to the list of their accepted root-certificates.
+type LogRoots map[string][]*x509.Certificate
+
+// ActiveLogs creates a new LogList containing only non-disqualified logs from the original.
+func (ll *LogList) ActiveLogs() LogList {
+	var active LogList
+	// Keep all the operators.
+	active.Operators = deepcopy.Copy(ll.Operators).([]Operator)
+	for _, l := range ll.Logs {
+		if l.DisqualifiedAt <= 0 || time.Since(time.Unix(int64(l.DisqualifiedAt), 0)) < 0 {
```

Comment:
nit: `time.Until(...) > 0` might be a little easier to read, since that translates to "there is time left until it is disqualified", rather than "there is no time left since it was disqualified".

Reply:
Updated.